File: 2025/07/prog_42746_가장큰수.js
/*
문자열로 정렬 
6, 10, 2 => 6, 2, 10
3, 30, 34, 5, 9 => 9, 5, 34, 30, 3
*/
function solution(numbers) {  
    const strings = numbers.map((n) => String(n))
    strings.sort((a, b) => (b+a).localeCompare(a+b)) // 3+30=330, 30+3=303
    const result = strings.join("");
    
    return result[0] === "0" ? "0" : result;
}

File: 2026/01/prog_42746_가장큰수.py
# 34, 30, 3 -> 3434, 3030, 33
# 547, 54, 5 -> 547547, 5454, 55 => 5, 547, 54
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*4, reverse=True)
    
    return str(int("".join(numbers)))